pandas/core/reshape.py

```diff
@@ -405,6 +406,9 @@ def _slow_pivot(index, columns, values):
 
 
 def unstack(obj, level, fill_value=None):
+    if fill_value:
+        validate_fill_value(fill_value, obj.values.dtype)
```

Comment:
`obj` may be a `DataFrame` AFAIK. I call `values` to get the `numpy` array here to consolidate the `dtype` (which means that e.g. a `DataFrame` with columns of mixed type will accept `fill_value` according to `object` rules). Is there a way to get this without accessing the underlying array directly?